finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -37,6 +37,13 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
   private[this] val closed: AtomicBoolean = new AtomicBoolean(false)
   protected[this] def isClosed = closed.get
 
+  private[this] val closedId: AtomicInteger = new AtomicInteger(0)
+  @tailrec private[this] def addClosedId(id: Int): Unit = {
+    val i = closedId.get
+    val max = if (id > i) id else i
+    if (!closedId.compareAndSet(i, max)) addClosedId(id)
+  }
```

Comment:
On second, thought, is this safe?

```
t0: closedId.get = 1
t1: closedId.get = 1
t1: max = 4
t0: max = 2
t0: cas(2, 1): true
t1: cas(4, 1): false
t2: closedId.get = 2
...
```

Maybe we don't actually care in this case, but it seems possible to miscalculate the max id